packages/frontend/web-ui/src/game/components/GameListItem.jsx

```diff
@@ -0,0 +1,40 @@
+import { Link as RouterLink } from 'react-router-dom';
+import { Button, Typography, Link, Box } from '@mui/material';
+import { JoinInnerOutlined, Share } from '@mui/icons-material';
+
+export const GameListItem = ({ game }) => {
+  return (
+    <Box component="div" className="game-list-item">
+      <Typography variant="h5" className="game-list-item-text" component="h5">
+        {game.name === undefined ? '--' : game.name}
+      </Typography>
+      {game.state.status === 'nonStarted' ? (
+        <Box component="div">
+          <Link component={RouterLink} to="/">
+            <Button
+              type="button"
+              className="game-list-item-button"
+              variant="contained"
+              startIcon={<Share />}
+            >
+              Share
+            </Button>
+          </Link>
+        </Box>
+      ) : (
+        <Box component="div">
+          <Link component={RouterLink} to="/">
+            <Button
+              type="button"
+              className="game-list-item-button"
+              variant="contained"
+              startIcon={<JoinInnerOutlined />}
+            >
+              Join
+            </Button>
+          </Link>
+        </Box>
```

Comment:
I think you're assuming games are either non started or active but, what if the game is finished? Remember this is an agnostic component, so it might be the case in the future. For now I would render an empty Box in that case.